pyron-plugin-impl/src/main/scala/com/cloudentity/pyron/plugin/impl/transform/response/TransformResponsePlugin.scala

```diff
@@ -37,14 +38,16 @@ class TransformResponsePlugin  extends ResponsePluginVerticle[TransformerConf] {
 
   override def apply(ctx: ResponseCtx, conf: TransformerConf): Future[ResponseCtx] = Future.successful {
     val jsonBodyOpt: Option[JsonObject] = parseJsonBodyIfRequired(ctx, conf)
-    ctx |>
-      transformResJsonBody(resolveBodyOps(ctx, conf.body, jsonBodyOpt), jsonBodyOpt) |>
-      transformResHeaders(resolveHeaderOps(ctx, conf.headers, jsonBodyOpt))
+
+      ctx |> transformResJsonBody(resolveBodyOps(ctx, conf.body, jsonBodyOpt), jsonBodyOpt) |>
+      transformResHeaders(resolveHeaderOps(ctx, conf.headers, jsonBodyOpt)) |>
+      transformHttpStatus(conf.status)
   }
 
   // to avoid parsing body to JsonObject over and over again we do it only if JSON body reference exists or body operations are set
   def parseJsonBodyIfRequired(ctx: ResponseCtx, conf: TransformerConf): Option[JsonObject] = {
-    if (conf.parseJsonBody) {
+    if(emptyApiResponseBody(ctx)) Some(new JsonObject())
```

Comment:
wouldn't that change response even if none body change is defined?
I mean lets suppose we want to add header to empty response body. Wouldn't new impl return empty json insead of empty body to client?

Reply:
this `parseJsonBodyIfRequired` function is called only if `rawConf.body.nonEmpty || jsonBodyRefExists(rawConf),`, so if you wouldn't like to change body(based on plugin config), it should not change body.

Reply:
ok